llama_stack/providers/impls/meta_reference/inference/inference.py

```diff
@@ -87,6 +92,7 @@ async def completion(
             logprobs=logprobs,
         )
         self.check_model(request)
+        request = await request_with_localized_media(request)
```

Comment:
This seems to break if client uploads image via URL. 

```
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 421, in request_with_localized_media
    m.content = await _convert_content(m.content)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 415, in _convert_content
    return [await _convert_single_content(c) for c in content]
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 415, in <listcomp>
    return [await _convert_single_content(c) for c in content]
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 408, in _convert_single_content
    url = await convert_image_media_to_url(content, download=True)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/utils/inference/prompt_adapter.py", line 77, in convert_image_media_to_url
    r = await client.get(media.image.uri)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 1814, in get
    return await self.request(
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 1572, in request
    request = self.build_request(
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 346, in build_request
    url = self._merge_url(url)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 376, in _merge_url
    merge_url = URL(url)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_urls.py", line 117, in __init__
    self._uri_reference = urlparse(url, **kwargs)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_urlparse.py", line 158, in urlparse
    raise InvalidURL("URL too long")
httpx.InvalidURL: URL too long
```

Reply:
@yanxi0830 I think the image becomes too large and is shipped as a "data" URL. This seems important to resolve. Filing an issue.